src/module.ts

```diff
@@ -188,27 +188,6 @@ const setSecurityResponseHeaders = (nuxt: Nuxt, headers: SecurityHeaders) => {
   }
 }
 
-const setSecurityRouteRules = (nuxt: Nuxt, securityOptions: ModuleOptions) => {
```

Comment:
question: shouldn't we also define the same approach for line 174 -> `setSecurityResponseHeaders`?

Reply:
Yes, the same.

Reply:
I have made the changes to allow setting `headers` at runtime. However, objects cannot be overriden by matching environment variables. This may be related to Nuxt.

```ts
NUXT_SECURITY_HEADERS_CROSS_ORIGIN_RESOURCE_POLICY=false // Work
NUXT_SECURITY_HEADERS_CONTENT_SECURITY_POLICY=false // Do not work
```

Reply:
If objects cannot be ovewritten, then a lot of functionality wont benefit from this change. Almost all middlewares and several headers are configured via objects (CSP, PermissionsPolicy, StrictTransportPolicy, etc).

I am not sure about this change then. In general the idea is really good, but if we are changing it and we cannot provide the same functionality as it is right now (supporting object configuration) then this can lead to bugs and issues. 

I am affraid we would need to wait for the time when Nuxt core would solve the issue you mentioned.

Reply:
I totally agree. 

The changes here requires the ability to change route rules at runtime, which is not feasible with Nitro and the ability to override config objects with environment variables, which is not feasible with Nuxt. I hope there would be a native solutions to these in the future.

Anyway, thanks for your time and efforts on this module.

Reply:
@becem-gharbi would you be ok with leaving this PR open for now until a proper fix will be released?

This is a great feature for sure, I just dont want to ship something that solves one issue and introduce another. Once the fix will be done, I am completely into looking at this once again and merging when ready :)

Reply:
Also, could you create an issue in Nuxt about it and link it here? It will be easier to keep track of changes and start working on it once released :)

Reply:
I have created the issue, https://github.com/nuxt/nuxt/issues/23695